lib/public/utilities/DataSource.js

```diff
@@ -0,0 +1,117 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+import { Observable } from '/js/src/index.js';
+import { ObservableData } from './ObservableData.js';
+
+/**
+ * Class to abstract notifying from data changes
+ * @template T - type of data served by data source
+ */
+export class DataSource extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+        this._dataSource = null;
+
+        /**
+         * Produce default (initial) value stored in ObservableData
+         * @return {*} default
+         */
+        this._default = () => null;
+
+        /**
+         * @type {ObservableData<T>}
+         * @private
+         */
+        this._observableData = new ObservableData(null);
+    }
+
+    /**
+     * Fetch data
+     * @param {*} _args arguments used by this data sources or underlying one
+     * @return {Promise<void>} resolves once the data fetching has ended
+     * @abstract
+     */
+    async fetch(_args) {
```

Comment:
Single responsibility principle & Interface segregation: datasource has no reason to have a fetch function. You may have a `ConstantDataSource` that provides always the same value, which is a constant for example.
In my example note that the only thing in `DataSource` is `getObservableData` (https://www.typescriptlang.org/play?ssl=34&ssc=5&pln=34&pc=22#code/JYOwLgpgTgZghgYwgAgPICMDO0Bud0A2EAInGHADwAqAfMgN4BQyLyCArlFBOAFzJUA3I2asQAezDAYAT34AKAJTIAvHRzjgAE2GiW4rLggKEcAgXSIA1guVrkG7XfWadjAL4iEBOJkzIAQXYwcQxsKDxCEjJKWmRgAFsAByIEnjB-MNx8IlJyajomVmQkqGA8SGQAfQ4udP4hEWKEcRBMMCh2BBCoeVBgKTMANTN2YwFlIuLWMAALYEwAOhrObnBVeJAB4GHRiGFizz1kAHMIMDZV9KUGY+LuME4QZDmF5dq1sAPWI+LsC4+10BfAmt2mM3mSxWdXWKkuMK+x1+Ykk0hkNym4IA9FjkAA5VGyZAGcI4aBLRaLJFNVgkowYu6sHHIbgnBaQKDEwwRaDUo6gDnwJDIPJwADK4k4SAKYNYZzAWQiOWi5CU-EVkVyMQKwiO3l8-gAShAEpIVbFCscWm0Ol0evIknAZARxHAtAB+BqTanHZkATUlyCsEgA7vELmZuG6ZAAaZAASQA5OZkJhEik0cd2mRgAhkC63aATgU1chjabIKKZZjpg8nsgQBAw+WzaKlN8WJ5ilnyFI85gukg-CXHc7XVovfwW5XtXEa-dzvXG82Ta2Yg6nQWtIoO8hkSw4EkUjIKGKaPJTNhMPx6GLBxA-LZVHQxXGADLjotP+xiveKfi-vQyDMmK6YENIwAQFoyBwPmkjILM0AoCECFwGSqbiGkrwgCcyAhuIUBWEWjIsHWUDPJeD6LHeCBDpg7ZIrqXg+H4ZarjO5ASlKEAymBJrpP4opcVA0rTuaBSWsUpTlGQKBRlorQEDIXKksqorqtymriWJVa0DQujNK07SdN0BEMuCLDMvGqZgWicYDKmsySgQ0HoCgEgRpgVhQcSMBsJKUDYCRLyQosdJKlEoobMugTBKEmlqeuOkxIsA60Q+9EgOw5iKDuPrHL4MggHmW6lgAClAmELDxjhaJJFmvEs4VaaKizAhccLJeQixbkWDHBcyABi5wILMIUoFoMQ0uCjVhQlkUpe1GxdXAqX3n48j0O4MH+FQig+sU8oaolqrehZLKLuRIVvM1J1wLunhHGAMhJCgZVwGyICyVoK0yp17HiTWWFwPwBgAFYQN0MbHJN5BTgDukANoALo0B4aOMAK0BCm9H2gN9K3CdKcQQAAHpAIBaIJMREzx72fQTCOzjQ9UsKV4gwDA-z8FlCRuVAcbgQkAw8+wfPQP+yAVVV2AULV+keMxBrIAAEmAYBJPT+OQFoQkBcTdB8Wk4D+FrX064T+s8XOxzSRUckQG6inKbdC1w2g81avkv1UCjLMGaw1rGXaZlnQ1oWu17sFwjFQQhMdbtwPIK09Z+OFKHlhzTTtRUleO8gc1z5yC4kAxh+CRAXLD4pW-DFbmrT1B+8gAA+DbZQQ0VNmx9d69xTeowx53V7TixHZ75pKHNpIQPINz2POFnSMg8gAIQj1byhkSAgjAbimBOdl0GNmSnKzIer0gMFxSzZH5ptVcsLIBv3Fj+cCdR1P7WLBfSkUD7ftzyL3OiwGidFvx0BTmlOi8gUaKDjMyFWSFkAyEDApca-liqyXNsAVocZ2BJFhigR0ZwYKUwwsSK+ICLIfkLDhCBPc1zdV6unZQiDkGoPYDZZISkXhQDgG0GA0AMHQEqlAa+PxM4WXcHlAauJaHQXYGmHCvki4XAAGT5lLhcYKQcq40ytgonycIX4iQgKnN0Q8LKmKQEYrQiw5g8Dns4Z+BjuLRQ7juPeIpxAYSwvMFR4FvIhTIC8Xx3hHbPAgKfGQ2ETjBW3q4zihjxxQQescY449VKJ3MhZRJt8J6inSXqFi-gADCiBEJaDNozXubizG8R4fxE2Us8bmygpbfuNspJlHthdJ2IBeGmFGuMX2yNdx21kv0hSgyXaFJiBpbJUd-5M3yGM-22c9EmXtOXaYBSlnmi7mGOO8UDlthTiwk4Gd0kFUwLnOCljC7-BLsLMAuzijMjKYhBAVhNgYOGYheIfkQwoHPuhAYciEzAtBQMOMiSIXnWZAAUQINgOMWgIBEBOFMlCsE1YaxqTrPuZiDpynfvM06spwT5IjhS+6fIRDMgJGGUA-zwLpBQYGBITpULoUwJhc4ATcJBOMF4IyVcFgpCdAoGx4xCUdNWTXfuvN+Y0BccZIsGx5CypcTWRJsq34KjpV-R+YAf5HiUptYKYCMoMIAERRRgJKSmq87XQ3Ogor8K8XF2oAOooDgNwB5Wgix2qRPtdwQA)